keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlUnhappyVetoRepository.kt

```diff
@@ -31,14 +32,30 @@ class SqlUnhappyVetoRepository(
   private val sqlRetry: SqlRetry
 ) : UnhappyVetoRepository(clock) {
 
-  override fun markUnhappyForWaitingTime(resourceId: String, application: String, wait: Duration) {
+  override fun markUnhappyForWaitingTime(resourceId: String, application: String, wait: Duration?) {
     sqlRetry.withRetry(WRITE) {
       jooq.insertInto(UNHAPPY_VETO)
         .set(UNHAPPY_VETO.RESOURCE_ID, resourceId)
         .set(UNHAPPY_VETO.APPLICATION, application)
-        .set(UNHAPPY_VETO.RECHECK_TIME, calculateExpirationTime(wait).toTimestamp())
+        .run {
+          calculateExpirationTime(wait)?.toTimestamp().let { expiryTime ->
+            if (expiryTime == null) {
+              setNull(UNHAPPY_VETO.RECHECK_TIME)
+            } else {
+              set(UNHAPPY_VETO.RECHECK_TIME, expiryTime)
+            }
+          }
```

Comment:
```suggestion
          calculateExpirationTime(wait)?.let { expiryTime ->
            if (expiryTime == null) {
              setNull(UNHAPPY_VETO.RECHECK_TIME)
            } else {
              set(UNHAPPY_VETO.RECHECK_TIME, expiryTime.toTimestamp())
            }
          }
```

Reply:
I like that better, but it doesn't fix the test!